fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -645,6 +645,12 @@ public ComputeNode getComputeNodeWithBePort(String host, int bePort) {
     public List<Long> getComputeNodeIds(boolean needAlive) {
         ImmutableMap<Long, ComputeNode> idToComputeNode = idToComputeNodeRef;
         List<Long> computeNodeIds = Lists.newArrayList(idToComputeNode.keySet());
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
+            computeNodeIds = warehouse.getAnyAvailableCluster().getComputeNodeIds().
```

Comment:
`getComputeNodeIds()` returns `List<Long>`, no need to filter through stream again.